extensions/typescript-language-features/package.json

```diff
@@ -982,6 +982,22 @@
           "markdownDescription": "%typescript.preferences.includePackageJsonAutoImports%",
           "scope": "window"
         },
+        "typescript.preferences.autoImportFileExcludePatterns": {
```

Comment:
Does it make sense to have two settings or a single `js/ts` setting?  (I know this is how most of our existing settings work, just curious)

Reply:
Well, the setting is actually file-level, and I think there are some advantages to that (it means if you have a Workspace for a monorepo, you could configure the auto-imports of each package differently even though there’s only one TS Server instance). Unfortunately, historical precedent dictates that `typescript.preferences` only apply to TS files and `javascript.preferences` only apply to JS files. So while I can’t think of a reason why you’d want to configure this differently between JS and TS, if we made it just `typescript.preferences`, it wouldn’t be usable in JS files unless we hard-coded an exception for it. It’s especially weird because this strategy kind of defeats the language overrides possible in settings.json files. I think we should leave this one matching all the others, but I would love to discuss how we can clean up this whole scheme sometime.